examples/timeseries/event_classification_for_payment_card_fraud_detection.py

```diff
@@ -0,0 +1,430 @@
+"""
+
+feature_set_3 = tp.glue(
+    transactions_per_terminal.calendar_hour(),
+    transactions_per_terminal.calendar_day_of_week(),
+)
+
+"""
+Finally, we group together all the features and the label.
+"""
+
+all_data = tp.glue(
+    transactions_per_terminal, feature_set_1, feature_set_2, feature_set_3
+).drop_index()
+
+print("All the available features:")
+all_data.schema.feature_names()
+
+"""
+We extract the name of the input features.
+"""
+
+input_feature_names = [
+    k for k in all_data.schema.feature_names() if k.islower()
+]
+
+print("The model input features:")
+input_feature_names
+
+"""
+For neural networks to work correctly, numerical input must be normalized. A common approach is to apply z-normalization, which involves subtracting the mean and dividing by the standard deviation estimated the training data to each value. In forecasting, such z-normalization is not recommended as it would lead to future leakage. Specifically, to classify a transaction at time t, we cannot rely on data after time t since, at serving time when making a prediction at time t, no subsequent data is available yet. In short, at time t, we are limited to using data that precedes or is concurrent with time t.
+
+The solution is therefore to apply z-normalization **over time**, which means that we normalize each transaction using the mean and standard deviation computed from the past data **for that transaction**.
+
+Future leakage is pernicious. Luckily, Temporian is here to help: The only operator that can cause future leakage is `tp.leak`. If you are not using `tp.leak`, your preprocessing is **guaranteed** not to contain future leakage.
+
+**Note:** For advanced pipelines, you can also check programatically that a feature does not depends on a `tp.leak` results.
+"""
+
+# Cast all values (e.g. ints) to floats.
+values = all_data[input_feature_names].cast(tp.float32)
+
+# Apply z-normalization overtime.
+normalized_features = (
+    values - values.simple_moving_average(math.inf)
+) / values.moving_standard_deviation(math.inf)
+
+# Restore the original name of the features.
+normalized_features = normalized_features.rename(values.schema.feature_names())
+
+print(normalized_features)
+
+"""
+One issue is that the first transactions will be normalized using poor estimates of the mean and standard deviation since there are only a few transactions before them. So, we will remove the first week of data from the training dataset.
+
+Notice that the first values contain NaN. In Temporian, NaN represents missing values, and all operators handle them accordingly. For instance, when calculating a moving average, NaN values are not included in the calculation and do not generate a NaN result.
+
+However, neural networks are not good at handling NaN natively. So, we replace them with zeros.
```

Comment:
However, neural networks cannot natively handle NaN values. So, we replace them with zeros.

Reply:
*Done